src/behave/behaveTestRunner.ts

```diff
@@ -0,0 +1,209 @@
+        this.logger.log('info', `Running behave with locations: ${testRunArguments.locations.join(', ')}`);
+
+        const params = [ ...testRunArguments.argumentsToPass, ...testRunArguments.locations];
+
+        const result = await this.runBehave(config, additionalEnvironment, params).complete();
+        const states = parseTestStates(result.output);
+        if (empty(states)) {
+            // maybe an error occured
+            this.logger.log('warn', 'No tests run');
+            this.logger.log('warn', 'Output: ${result.output}');
+        }
+
+        return states;
+    }
+
+    private runBehave(config: IWorkspaceConfiguration, env: IEnvironmentVariables, args: string[]): IProcessExecution {
+        const behavePath = config.getBehaveConfiguration().behavePath();
+        if (behavePath === path.basename(behavePath)) {
+            this.logger.log('info', `Running ${behavePath} as a Python module`);
+            return runModule({
+                pythonPath: config.pythonPath(),
+                module: config.getBehaveConfiguration().behavePath(),
+                environment: env,
+                args,
+                cwd: config.getCwd(),
+                acceptedExitCodes: BEHAVE_NON_ERROR_EXIT_CODES,
+            });
+        }
+
+        this.logger.log('info', `Running ${behavePath} as an executable`);
+        return runProcess(
+            behavePath,
+            args,
+            {
+                cwd: config.getCwd(),
+                environment: env,
+                acceptedExitCodes: BEHAVE_NON_ERROR_EXIT_CODES,
+            });
+    }
+
+    private async loadEnvironmentVariables(config: IWorkspaceConfiguration): Promise<IEnvironmentVariables> {
+        const envFileEnvironment = await EnvironmentVariablesLoader.load(config.envFile(), process.env, this.logger);
+
+        const updatedPythonPath = [
+            config.getCwd(),
+            envFileEnvironment.PYTHONPATH,
+            process.env.PYTHONPATH,
+            DISCOVERY_OUTPUT_PLUGIN_INFO.PACKAGE_PATH
+        ].filter(item => item).join(path.delimiter);
+
+        const updatedBehavePlugins = [
+            envFileEnvironment.BEHAVE_PLUGINS,
+            DISCOVERY_OUTPUT_PLUGIN_INFO.MODULE_NAME
+        ].filter(item => item).join(',');
+
+        return {
+            ...envFileEnvironment,
+            PYTHONPATH: updatedPythonPath,
+            BEHAVE_PLUGINS: updatedBehavePlugins,
+        };
+    }
+
+    private getDiscoveryArguments(rawBehaveArguments: string[]): IBehaveArguments {
+        const argumentParser = this.configureCommonArgumentParser();
+        const [knownArguments, argumentsToPass] = argumentParser.parse_known_args(rawBehaveArguments);
+        return {
+            locations: (knownArguments as { locations?: string[] }).locations || [],
+            argumentsToPass: ['-d', '-f', 'json', '--no-summary', '--no-snippets'].concat(argumentsToPass),
+        };
+    }
+
+    // @ts-expect-error
```

Comment:
Can this be removed?

Reply:
Yes, should be doable. Currently the test parameter is not used, but it should be.